system/jlib/jstats.cpp

```diff
@@ -1672,6 +1763,114 @@ bool CRuntimeStatisticCollection::serialize(MemoryBuffer& out) const
 }
 
 
+//---------------------------------------------------------------------------------------------------------------------
+
+void CRuntimeSummaryStatisticCollection::DerivedStats::mergeStatistic(unsigned __int64 value, unsigned node)
+{
+    if (count == 0)
+    {
+        min = value;
+        max = value;
+        minNode = node;
+        maxNode = node;
+    }
+    else
+    {
+        if (value < min)
+        {
+            min = value;
+            minNode = node;
+        }
+        if (value > max)
+        {
+            max = value;
+            maxNode = node;
+        }
+    }
+    count++;
+    double dvalue = (double)value;
+    sumSquares += dvalue * dvalue;
+}
+
+CRuntimeSummaryStatisticCollection::CRuntimeSummaryStatisticCollection(const StatisticsMapping & _mapping) : CRuntimeStatisticCollection(_mapping)
+{
+    derived = new DerivedStats[ordinality()+1];
+}
+
+CRuntimeSummaryStatisticCollection::~CRuntimeSummaryStatisticCollection()
+{
+    delete[] derived;
+}
+
+void CRuntimeSummaryStatisticCollection::ensureNested()
+{
+    if (!nested)
+        nested = new CNestedSummaryRuntimeStatisticMap;
+}
+
+void CRuntimeSummaryStatisticCollection::mergeStatistic(StatisticKind kind, unsigned __int64 value)
+{
+    CRuntimeStatisticCollection::mergeStatistic(kind, value);
+    unsigned index = queryMapping().getIndex(kind);
+    derived[index].mergeStatistic(value, 0);
+}
+
+void CRuntimeSummaryStatisticCollection::recordStatistics(IStatisticGatherer & target) const
+{
+    CRuntimeStatisticCollection::recordStatistics(target);
+    for (unsigned i = 0; i < ordinality(); i++)
+    {
+        DerivedStats & cur = derived[i];
+        if (cur.count)
+        {
+            StatisticKind kind = getKind(i);
+            StatisticKind serialKind= querySerializedKind(kind);
+
+            unsigned __int64 minValue = convertMeasure(kind, serialKind, cur.min);
+            unsigned __int64 maxValue = convertMeasure(kind, serialKind, cur.max);
+            if (minValue != maxValue)
+            {
+                double sum = (double)convertMeasure(kind, serialKind, values[i].get());
+                //Sum of squares needs to be translated twice
+                double sumSquares = (double)convertMeasure(kind, serialKind, convertMeasure(kind, serialKind, cur.sumSquares));
```

Comment:
This implicitly casts from a double and may overflow an int.  It needs a different function.